src/main/resources/com/google/api/codegen/java/test.snip

```diff
@@ -55,14 +55,23 @@
     {@setResponse(test.mockResponse)}
 
     {@initCode(test.initCode)}
+
+    {@setResponse(test.mockResponse)}
+
     {@methodCall(test)}
 
     List<GeneratedMessage> actualRequests = serviceHelper.getService().getRequests();
     Assert.assertEquals(1, actualRequests.size());
     {@test.requestTypeName} actualRequest = ({@test.requestTypeName})actualRequests.get(0);
 
-    @join assert : test.asserts
-      Assert.assertEquals(actualRequest.{@assert.actualValueGetter}(), {@assert.expectedValueIdentifier});
+    @join assert : test.requestAsserts
+      Assert.assertEquals({@assert.expectedValueIdentifier}, \
+        actualRequest.{@assert.actualValueIdentifier}());
+    @end
+
+    @join assert : test.responseAsserts
+      Assert.assertEquals(expectedResponse.{@assert.expectedValueIdentifier}(), \
+        actualResponse.{@assert.actualValueIdentifier}());
```

Comment:
It's confusing for the names to be xIdentifier when they are actually get function call names.